### ISSUE TITLE
debug log sidebar dead object

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#816]
_[Trac time 20100106 050445Z]_ Uncertain if I can reproduce this, but I closed a window that had multiple tabs, then opened a new window by using the send to other instance method.

Produced a new window and loaded the new file, but every subsequent file load raised an exception:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14614, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/lib/pubsub.py", line 811, in sendMessage
    self.__topicTree.sendMessage(aTopic, message, onTopicNeverCreated)
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/lib/pubsub.py", line 498, in sendMessage
    deliveryCount += node.sendMessage(message)
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/lib/pubsub.py", line 336, in sendMessage
    listener(message)
  File "/home/rob/src/peppy/peppy/plugins/error_log.py", line 180, in showError
    if self.frame == frame:
  File "/usr/lib64/python2.5/site-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14564, in __getattr__
    raise PyDeadObjectError(self.attrStr % self._name)
PyDeadObjectError: The C++ part of the DebugLogSidebar object has been deleted, attribute access no longer allowed.
```

**Comment by robmcmullen** [comment]
_[Trac time 20100107 003249Z]_ Looks like it is reproduceable by closing a window regardless of the number of tabs

**Comment by robmcmullen** [comment]
_[Trac time 20100107 212218Z]_ 057547cad5f4be297121616fac08b79566b49256 Refs #816: added debug print and temporary message dialog in an attempt to pinpoint the problem

**Comment by robmcmullen** [comment]
_[Trac time 20100125 224327Z]_ I think it's related to not unsubscribing from pubsub messages when an object is deleted.

**Comment by robmcmullen** [comment]
_[Trac time 20100127 230858Z]_ ec313a548e1dd97053381acf6121b196eac7ae19 Fixed #816: added Sidebar.removeListeners method that's automatically called by **del**
- moved ErrorLogSidebar pubsub creation into createListeners added removeListeners
